packages/fxa-settings/src/lib/oauth/hooks.tsx

```diff
@@ -167,29 +189,49 @@ export function useFinishOAuthFlowHandler(
 
       let keys;
       if (integration.wantsKeys() && keyFetchToken && unwrapBKey) {
-        const { kB } = await authClient.accountKeys(keyFetchToken, unwrapBKey);
-        keys = await constructKeysJwe(
-          authClient,
-          oAuthIntegration,
-          accountUid,
-          sessionToken,
-          keyFetchToken,
-          kB
-        );
+        try {
+          const { kB } = await authClient.accountKeys(
+            keyFetchToken,
+            unwrapBKey
+          );
+          keys = await constructKeysJwe(
+            authClient,
+            oAuthIntegration,
+            accountUid,
+            sessionToken,
+            keyFetchToken,
+            kB
+          );
+        } catch (e) {
+          return tryAgainError();
```

Comment:
I don't see where we're handling these in Backbone. If you check the `signIn` function in the `signin-mixin`, you'll see a rethrow with a comment that we handle it elsewhere. If you search for `this.signIn`, you'll see we catch errors and run `onSignInError`, but we rethrow here as well.

This error message tells the user to close the tab and try again, which I think is exactly what we would want them to do in these scenarios because the most likely scenario is that something is wrong with the `redirect_to` or `state` query param. Secondarily there could have been a network request problem, in which case we could just say "try again" without closing the tab, but we're not really doing that anywhere else.